### ISSUE TITLE
your new code

### ISSUE THREAD

**Issue by Welsh-Boogie** [sagishporer/hashcode-2021-qualification#11]
you said u have a new code, would u mind explaining what are you bruteforcing? did u do like I was saying or something different? and would u mind sharing it here or pushing it?

**Comment by sagishporer** [comment]
For every intersection try to:
1. Switch between every pair of green lights - check if it improves.
2. Add/Remove time from every green light - check if it improves.

**Comment by Welsh-Boogie** [comment]
> For every intersection try to:
> 
> 1. Switch between every pair of green lights - check if it improves.
> 2. Add/Remove time from every green light - check if it improves.

hmmm second one seems would imrpove, cant believe that went oevr my head

**Comment by Welsh-Boogie** [comment]
how has it affected code running time? and could u attach it?

**Comment by sagishporer** [comment]
B takes about 30 minutes, C - 150 minutes, the others I don't have results yet. Currently trying E. On inputs D & F it's much harder to do brute force.

**Comment by Welsh-Boogie** [comment]
> B takes about 30 minutes, C - 150 minutes, the others I don't have results yet. Currently trying E. On inputs D & F it's much harder to do brute force.

hmmm if u want like I said i could run it, I'm sitting free and my up isnt being used, I'm now interested even more to see the code hmm

**Comment by Welsh-Boogie** [comment]
when code ill send u the output as in the a, b, c, d, e and f files generated and thing in terminal

**Comment by Welsh-Boogie** [comment]
> B takes about 30 minutes, C - 150 minutes, the others I don't have results yet. Currently trying E. On inputs D & F it's much harder to do brute force.

i understand, it has too many things to account for

**Comment by Welsh-Boogie** [comment]
i just had another idea, ill try doing those cases when the next push happens

**Comment by Welsh-Boogie** [comment]
@sagishporer u pushing it man? i can leave it running rn im not sure if ill be available later

**Comment by Welsh-Boogie** [comment]
@sagishporer  THIS GUY IS COPYING YOUR WORK AND CLAIMING IT IS HIS WORK, STOP HIM MAN

https://github.com/afikmenashe21/Google-hashcode-practice

**Comment by sagishporer** [comment]
Thank you for your concern. Weird. He can fork the code and continue with the Apache license.

I'm not going to invest time in handling that. It's for fun, not business.

**Comment by Welsh-Boogie** [comment]
> Thank you for your concern. Weird. He can fork the code and continue with the Apache license.
> 
> I'm not going to invest time in handling that. It's for fun, not business.

yeah, sorry, I'm just annoyed at people who do such things, the only ones they are fooling is themselves. they just show that they aren't capable of doing things themselves



anyhow how are tests going?

**Comment by Welsh-Boogie** [comment]
@sagishporer u have an estimate of when ur going to release or when i can take a look at it?

**Comment by sagishporer** [comment]
Not sure. I don't want the extended round to get flooded.

**Comment by Welsh-Boogie** [comment]
> Not sure. I don't want the extended round to get flooded.

well can I take look at it lol, I wont distribute lol cuz people mostly don't care abt code, they just care abt score when this competition is just for fun

**Comment by Welsh-Boogie** [comment]
> Not sure. I don't want the extended round to get flooded.

when u push the code ill show u how u can keep skidders away

**Comment by Welsh-Boogie** [comment]
> Not sure. I don't want the extended round to get flooded.

@sagishporer if u privately want to share u could share thru email 0-0, XD sorry I'm just too excited and impatient, its my first hashcode and my teammates are useless

**Comment by Welsh-Boogie** [comment]
@sagishporer could u confirm if u can share the src in the next 2-4 hours so I can see (the code not output) for what u have done in green light cycle and the new brute-forcing u were talking abt, also I head some people saying ratios in randomization may increase score and I also have a way to speed up bruteforcing.  a yes or no would be nice, I want to test before the competition gets over on kraggle, also did ur team qualify?

**Comment by Welsh-Boogie** [comment]
@sagishporer if u don't want to make it public u can privately send it at secondisland222@gmail.com with files attatached thanks

**Comment by Welsh-Boogie** [comment]
ill be back in 16 hours or less less actually and ill tell u what i was thinking about

**Comment by Welsh-Boogie** [comment]
im back from exam, u there @sagishporer ?

**Comment by sagishporer** [comment]
I don't have time to put everything in order. I'll just clean the code and push it tonight or tomorrow.

**Comment by Welsh-Boogie** [comment]
> I don't have time to put everything in order. I'll just clean the code and push it tonight or tomorrow.

its fine man don't worry, send the individual one or any order, ill clean and add comments for u @sagishporer

**Comment by sagishporer** [comment]
I've pushed all the code.

**Comment by Welsh-Boogie** [comment]
> I've pushed all the code.

k lemme finish my food and ill start cleaning it up and adding comments, ill do proper spacing and things too

**Comment by Welsh-Boogie** [comment]
> I've pushed all the code.



> I've pushed all the code.

what is


            // Hill-Climbing
            //solution = OptimizeBruteForce(problem, solution, int.MaxValue);

**Comment by Welsh-Boogie** [comment]
its going in loops right now and I'm trying to understand will get back when i understand it

**Comment by Welsh-Boogie** [comment]
its been an hour for it, ill check if its looped or not

**Comment by sagishporer** [comment]
Again this is all the code for the latest version. There different configuration. And yes, not all the methods & optimizations are suitable for all the inputs.

**Comment by Welsh-Boogie** [comment]
> Again this is all the code for the latest version. There different configuration. And yes, not all the methods & optimizations are suitable for all the inputs.

yes i  figured most of the things out, its amazing what u have done,  ill download the calculator someone made on github and tell u!  its just taking too long to run also, I'm making it so it doesn't quit cycle simulation as soon as it gets one good result, so it gets best results for cycle

**Comment by Welsh-Boogie** [comment]
> Again this is all the code for the latest version. There different configuration. And yes, not all the methods & optimizations are suitable for all the inputs.

what is value for D and E? i was writing it in separate files as u did and i don't know which values to put there for D and E in ur latest version, ill put the old ones otherwise

**Comment by sagishporer** [comment]
The configuration stays as before (check the branched). All inputs but D can use brute force method. D can not (too slow).

**Comment by Welsh-Boogie** [comment]
> The configuration stays as before (check the branched). All inputs but D can use brute force method. D can not (too slow).

ill still run D with bruteforce, i can run it for days

**Comment by Welsh-Boogie** [comment]
i am running D bruteforce with 10 as parameter and running E bruteforce with the whole parameter thing removed

**Comment by Welsh-Boogie** [comment]
Hi ur instructions for running are bit unclear, could u clarify how u want me to run D and E and F?

**Comment by Welsh-Boogie** [comment]
I think i got F, i still need help understanding how u want me to make E and D

**Comment by Welsh-Boogie** [comment]
yep i need ur help, its been several hours running codes and only B worked, I'm trying to find what went wrong, I split it into sections and sections seem to work

**Comment by Welsh-Boogie** [comment]
how many times does loop work for each input, like its taken over 12 hours and they haven't been solved, how many loops for each? and how much time estimated?

**Comment by Welsh-Boogie** [comment]
making optimizations to run D with bruteforce as much as I can, do u already have outputs for other inputs? if u do send so I can free up slots and use them for bruteforcing D itself, hopefully cuts I made will work.

**Comment by Welsh-Boogie** [comment]
actually dont push the output, or else people who are not interested in code will copy it

**Comment by sagishporer** [comment]
The run time for B, C, E, F is up to 12 hours. I have all those outputs. The run time for D is much higher. I'm running only D now.

I'll publish the outputs after the extended round will be over, although anyone can just run the code on his computer.

**Comment by Welsh-Boogie** [comment]
> The run time for B, C, E, F is up to 12 hours. I have all those outputs. The run time for D is much higher. I'm running only D now.
> 
> I'll publish the outputs after the extended round will be over, although anyone can just run the code on his computer.

i made some changes to the code and am running D too, it probs wont fully run if my estimate is correct even if I run continuously so I'm trying to push it thru a bot net, hopefully, it works

**Comment by Welsh-Boogie** [comment]
> The run time for B, C, E, F is up to 12 hours. I have all those outputs. The run time for D is much higher. I'm running only D now.
> 
> I'll publish the outputs after the extended round will be over, although anyone can just run the code on his computer.

any new changes to code or is it finalized? I'm waiting for mine to run and ill tell u results. also the sorting thing

**Comment by sagishporer** [comment]
Haven't made any changes to the algorithm or to the sorting beyond the pushed code. 
I'm done.

I'll create brunches with the input specific configuration this weekend.

**Comment by Welsh-Boogie** [comment]
> Haven't made any changes to the algorithm or to the sorting beyond the pushed code.
> I'm done.
> 
> I'll create brunches with the input specific configuration this weekend.

whats ur latest code u just pushed?

**Comment by sagishporer** [comment]
Helper method for loading solution & checking it. Simple way to validate solution. 
Not used in the algorithm itself.

**Comment by Welsh-Boogie** [comment]
> Helper method for loading solution & checking it. Simple way to validate solution.
> Not used in the algorithm itself.

so basically no change in logic? ok I'm trying to make a sorter, could u also tell me what score outputs ur getting for each

**Comment by Welsh-Boogie** [comment]
> Helper method for loading solution & checking it. Simple way to validate solution.
> Not used in the algorithm itself.

currectly I'm working on:

1. output for D
2. sorting
3. different approaches

could u send me:

1. score for each input u have gotten

**Comment by Welsh-Boogie** [comment]
> A - 2,002
> B – 4,570,266
> C – 1,314,353
> D – 2,595,312 (still working)
> E – 761,931
> F – 1,479,243

Noice, im workin g on D too, whats ur currect D project look like? I'm currently trying to run it on a victim with a good cpu
